File: v131016/TeeNew/DBChart_Summary.pas
unit DBChart_Summary;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls, DBTables,
  {$ENDIF}
  Base_DBChart, DB, TeeProcs, TeEngine, Chart, Series, DBChart;

type
  TDBChartSummary = class(TBaseDBChart)
    Button1: TButton;
    Series1: TBarSeries;
    Table1: TTable;
    Label1: TLabel;
    ComboBox1: TComboBox;
    ComboBox2: TComboBox;
    CheckBox1: TCheckBox;
    procedure Button1Click(Sender: TObject);
    procedure ComboBox1Change(Sender: TObject);
    procedure ComboBox2Change(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeDBSumEdit;

procedure TDBChartSummary.Button1Click(Sender: TObject);
begin
  TeeDBSummaryEditor(Self,Series1);
end;

procedure TDBChartSummary.ComboBox1Change(Sender: TObject);
begin
  // Change the grouping field...
  Case ComboBox1.ItemIndex of
    0: Series1.XLabelsSource:='EmpNo';
    1: Series1.XLabelsSource:='PaymentMethod';
    2: Series1.XLabelsSource:='#Year#SaleDate';
    3: Series1.XLabelsSource:='ShipVia';
    4: Series1.XLabelsSource:='Terms';
  end;

  // Set a different color for each selected grouping field
  Series1.SeriesColor:=GetDefaultColor(ComboBox1.ItemIndex);
end;

procedure TDBChartSummary.ComboBox2Change(Sender: TObject);
begin
  // Change the aggregation ( sum, count, etc... )
  Case ComboBox2.ItemIndex of
    0: Series1.YValues.ValueSource:='#Sum#AmountPaid';
    1: Series1.YValues.ValueSource:='#Count#AmountPaid';
    2: Series1.YValues.ValueSource:='#High#AmountPaid';
    3: Series1.YValues.ValueSource:='#Low#AmountPaid';
    4: Series1.YValues.ValueSource:='#Avg#AmountPaid';
  end;
end;

procedure TDBChartSummary.CheckBox1Click(Sender: TObject);
begin
  CheckTable(Table1);

  // Open - close the table
  Table1.Active:=CheckBox1.Checked;
end;

procedure TDBChartSummary.FormShow(Sender: TObject);
begin
  inherited;
  ComboBox1.ItemIndex:=3;
  ComboBox2.ItemIndex:=0;
end;

initialization
  RegisterClass(TDBChartSummary);
end.

File: TeeNew/Builder2007/DBChart_Summary.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "DBChart_Summary.h"
#include "TeeDBSumEdit.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base_DBChart"
#pragma resource "*.dfm"
TDBChartSummary *DBChartSummary;
//---------------------------------------------------------------------------
__fastcall TDBChartSummary::TDBChartSummary(TComponent* Owner)
        : TBaseDBChart(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TDBChartSummary::ComboBox1Change(TObject *Sender)
{
  switch (ComboBox1->ItemIndex)
  {
    case 0: Series1->XLabelsSource = "EmpNo"; break;
    case 1: Series1->XLabelsSource = "PaymentMethod"; break;
    case 2: Series1->XLabelsSource = "#Year#SaleDate"; break;
    case 3: Series1->XLabelsSource = "ShipVia"; break;
    case 4: Series1->XLabelsSource = "Terms"; break;
  }
  Series1->SeriesColor = GetDefaultColor(ComboBox1->ItemIndex);
}
//---------------------------------------------------------------------------

void __fastcall TDBChartSummary::ComboBox2Change(TObject *Sender)
{
  switch (ComboBox2->ItemIndex)
  {
    case 0: Series1->YValues->ValueSource = "#Sum#AmountPaid"; break;
    case 1: Series1->YValues->ValueSource = "#Count#AmountPaid"; break;
    case 2: Series1->YValues->ValueSource = "#High#AmountPaid"; break;
    case 3: Series1->YValues->ValueSource = "#Low#AmountPaid"; break;
    case 4: Series1->YValues->ValueSource = "#Avg#AmountPaid"; break;
  }

}
//---------------------------------------------------------------------------

void __fastcall TDBChartSummary::Button1Click(TObject *Sender)
{
  TDBChartSumEditor *Editor = new TDBChartSumEditor(this);
  try
  {
    Editor->Tag = int(Series1);
    Editor->BorderStyle = bsDialog;
    Editor->Caption = "Summary properties";
    Editor->ShowModal();
  }
  __finally
  {
    delete Editor;
  }
}
//---------------------------------------------------------------------------

void __fastcall TDBChartSummary::CheckBox1Click(TObject *Sender)
{
  CheckTable(Table1);
  Table1->Active = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TDBChartSummary::FormShow(TObject *Sender)
{
  ComboBox1->ItemIndex = 3;
  ComboBox2->ItemIndex = 0;
}
//---------------------------------------------------------------------------

